nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -256,9 +257,23 @@ BtNavigator::initializeGoalPose()
 void
 BtNavigator::onGoalPoseReceived(const geometry_msgs::msg::PoseStamped::SharedPtr pose)
 {
-  nav2_msgs::action::NavigateToPose::Goal goal;
-  goal.pose = *pose;
-  self_client_->async_send_goal(goal);
-}
+  auto is_action_server_ready =
+    self_client_->wait_for_action_server(std::chrono::seconds(5));
+  if (!is_action_server_ready) {
```

Comment:
This could be streamlined as:

```
if (! self_client_->wait_for_action_server(std::chrono::seconds(0.5))) {
  // words words
}

```

although note that std::chrono::seconds needs an int so use another macro like ms